### PR TITLE
infra: add set up for blog

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Adds a blog set up that will make it easy to publish new posts.

You can test this locally by running the `build_blog.sh` script from the directory of the script. You'll need to have hugo for this, which you can do in the same way as the Dockerfile:

```sh
RUN mkdir -p hugo-bin && \
    cd hugo-bin && \
    wget https://github.com/gohugoio/hugo/releases/download/v0.126.1/hugo_extended_0.126.1_linux-amd64.tar.gz && \
    tar -xzf hugo_extended_0.126.1_linux-amd64.tar.gz

ENV PATH="${PATH}:/hugo-bin/"
```

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
I'm not sure if we want to use `gunicorn`? `python3 -m http.server` seemed like an easy solution in this case, am not sure if there are any drawbacks.

**Comment by DavidKorczynski** [comment]
Screenshot 2024-05-21 000454

**Comment by DavidKorczynski** [comment]
Screenshot 2024-05-21 000534

**Comment by oliverchang** [comment]
> I'm not sure if we want to use `gunicorn`? `python3 -m http.server` seemed like an easy solution in this case, am not sure if there are any drawbacks.

Seems like overkill for just local rendering. We can just use http.server for now.

File: infra/build/blog/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    wget https://github.com/gohugoio/hugo/releases/download/v0.126.1/hugo_extended_0.126.1_linux-amd64.tar.gz && \
+    tar -xzf hugo_extended_0.126.1_linux-amd64.tar.gz
+
+ENV PATH="${PATH}:/hugo-bin/"
+
+COPY content hugo.toml /
+COPY build_blog.sh /build_blog.sh
+RUN /build_blog.sh
+
+CMD exec python3 -m http.server 8011 -d /oss-fuzz-blog/page/public
```
Comment (oliverchang) [inline infra/build/blog/Dockerfile:32]: For our infra, we probably want to build a simple Cloud Build integration that just dumps the static site to a GCS bucket instead. We can do this later.

File: infra/build/blog/build_blog.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+BASE=$PWD
+
+# Build the site
+mkdir oss-fuzz-blog
+cd oss-fuzz-blog
+
+hugo new site page
+cd page
+git init
+git submodule add https://github.com/luizdepra/hugo-coder.git themes/hugo-coder
```
Comment (oliverchang) [inline infra/build/blog/build_blog.sh:26]: Can we pin this  dependency, and pull this in the Dockerfile instead?

**Review by oliverchang — COMMENTED** [review]
thanks!

File: infra/build/blog/build_blog.sh (hunk 2 above)
Comment (DavidKorczynski) [reply infra/build/blog/build_blog.sh]: Done

File: infra/build/blog/Dockerfile (hunk 1 above)
Comment (DavidKorczynski) [reply infra/build/blog/Dockerfile:37]: Yep sounds good, would you be able to do this? `/oss-fuzz-blog/page/public` holds the static website content.

Comment (oliverchang) [reply infra/build/blog/Dockerfile:37]: Yep, we can handle this!

**Review by oliverchang — APPROVED** [review]

**Pull request merged**